Tools/CISupport/ews-build/steps.py

```diff
@@ -4623,3 +4623,108 @@ def doStepIf(self, step):
 
     def hideStepIf(self, results, step):
         return not self.doStepIf(step)
+
+
+class AddReviewerMixin(object):
+    def environment(self):
+        owners = self.getProperty('owners', [])
+        if not owners:
+            return dict(
+                GIT_COMMITTER_NAME='EWS',
+                GIT_COMMITTER_EMAIL=FROM_EMAIL,
+            )
+
+        contributors, _ = Contributors.load(use_network=False)
+        return dict(
+            GIT_COMMITTER_NAME=contributors.get(owners[0], {}).get('name', 'EWS'),
+            GIT_COMMITTER_EMAIL=contributors.get(owners[0], {}).get('email', FROM_EMAIL),
+            FILTER_BRANCH_SQUELCH_WARNING='1',
+        )
+
+    def reviewers(self):
+        reviewers = self.getProperty('reviewers_full_names', [])
+        if len(reviewers) == 1:
+            return reviewers[0]
+        if reviewers:
+            return f'{", ".join(reviewers[:-1])} and {reviewers[-1]}'
+        return 'NOBODY (OOPS!)'
+
+
+class AddReviewerToCommitMessage(shell.ShellCommand, AddReviewerMixin):
+    name = 'add-reviewer-to-commit-message'
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(AddReviewerToCommitMessage, self).__init__(logEnviron=False, **kwargs)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        base_ref = self.getProperty('github.base.ref', DEFAULT_BRANCH)
+        head_ref = self.getProperty('github.head.ref', DEFAULT_BRANCH)
+
+        gmtoffset = int(time.localtime().tm_gmtoff * 100 / (60 * 60))
+        date = f'{int(time.time())} {gmtoffset}'
+        self.command = [
+            'git', 'filter-branch', '-f',
+            '--env-filter', f"GIT_AUTHOR_DATE='{date}';GIT_COMMITTER_DATE='{date}'",
+            '--msg-filter', f'sed "s/NOBODY (OO*PP*S!*)/{self.reviewers()}/g"',
+            f'{head_ref}...{base_ref}',
+        ]
+
+        for key, value in self.environment().items():
+            self.workerEnvironment[key] = value
```

Comment:
why do we need this?

Reply:
`self.env` didn't seem to work in unit tests (and I assume the real thing?), this does.